### PR TITLE
Template UFW user rules

### LINKED ISSUE #220: ufw rules are not reset after rerunnning the play
**Describe the bug**
ufw rules are not reset when rerunning the play as mentioned here #217 

**To Reproduce**
1. Run the playbook once with coturn_enabled: true
2. Rerun it again with coturn_enabled: false
3. Coturn ufw rules are still there

**Expected behavior**
ufw rules should be reset with every run

**Ansible version**
  ansible 2.10.12
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/d3h/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.8.10 (default, Jun  2 2021, 10:49:15) [GCC 9.4.0]

### PR DESCRIPTION
**Pull request by ChrisKar96**
Alternative to #252, without removing UFW
closes #220

### CODE REVIEW

File: templates/ufw/user6.rules.j2 [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+### tuple ### allow any {{ bbb_coturn_port_tls }} ::/0 any ::/0 in
+-A ufw6-user-input -p tcp --dport {{ bbb_coturn_port_tls }} -j ACCEPT
+-A ufw6-user-input -p udp --dport {{ bbb_coturn_port_tls }} -j ACCEPT
+
+### tuple ### allow udp {{ bbb_coturn_min_port }}:{{ bbb_coturn_max_port }} ::/0 any ::/0 in
+-A ufw6-user-input -p udp -m multiport --dports {{ bbb_coturn_min_port }}:{{ bbb_coturn_max_port }} -j ACCEPT
+
+{% endif %}
+### tuple ### allow udp {{ bbb_freeswitch_min_port }}:{{ bbb_freeswitch_max_port }} ::/0 any ::/0 in
```
Comment (crosscodr) [inline templates/ufw/user6.rules.j2:46]: Hi @ChrisKar96, I think you forgot to remove the variable `bbb_freeswitch_min_port` also from the comments :)

Comment (ChrisKar96) [reply templates/ufw/user6.rules.j2]: Nice catch again! Fixed it. I figured that we didn't really need to have new variables for the freeswitch ports, so I opted to revert that change.

**Comment by ChrisKar96** [comment]
After the last force push, UFW is truly idempotent, meaning that not changing any rules results in 'OK' response from ansible, instead of 'changed'.

**Comment by BrutalBirdie** [comment]
I am not good with these firewall templates if someone can verify them I am happy to approve the changes.

**Comment by crosscodr** [comment]
Hi @ChrisKar96, could you please rebase this PR on current bbb/2.3 branch instead of merging it?
In this state of the PR, one can not easily see whats really new in this PR.
Thank you :)

**Comment by ChrisKar96** [comment]
> 
> 
> Hi @ChrisKar96, could you please rebase this PR on current bbb/2.3 branch instead of merging it? In this state of the PR, one can not easily see whats really new in this PR. Thank you :)

Sorry about that, I was tired last night and did a fast forward without realizing. I fixed it now.

**Review by BrutalBirdie — APPROVED** [review]
Tested this PR and deployed to https://meet.bigbluebutton.dev/
All working produced an idempotent state for ufw.

+1
LGTM! Nice Job ❤️

**Comment by ChrisKar96** [comment]
As @BrutalBirdie was reviewing this, he discovered a breaking change that I hadn't thought of. So, right now with the PR, any role executed before BigBlueButton which sets ufw rules gets reset since BBB is templating the user.rules directly instead of adding them via the ufw commands. This means that rules applied before or after executing the role will only survive until the next execution of the role. The rules applied after the role break the idempotency of UFW task, while rules applied before are overwritten.

**Review by BrutalBirdie — COMMENTED** [review]
In discussion because of the breaking change.

**Comment by BrutalBirdie** [comment]
A Role should never have the directive to reset firewall changes done before him, since the Role can't know if the rules are valid or not.
Only the rules provided by the role itself should be added, changed or deleted by the role.
The addition of the UFW to the Role was also just the intention to add some basic security for copy-cats who just use this Role alone and don't know any better.

More so Greenlight and Coturn should be remove into separate Roles since they are not BBB specific. (Yes Greenlight kind off but its not needed for BBB to work.)

The Issue [#220](https://github.com/ebbba-org/ansible-role-bigbluebutton/issues/220) is correct but they way of templating the user.rules file is overreaching its boundaries.

Any thoughts? Please get involved :D

**Comment by BrutalBirdie** [comment]
Sorry, I have struggled with this PR a lot and came to the conclusion that we should not do it this way.

**Pull request closed without merging**